File: go/0001.two-sum/solution.go
// Created by ericklucioh at 2026/05/05 21:09
// leetgo: dev
// https://leetcode.com/problems/two-sum/

package main

import (
	"bufio"
	"fmt"
	"os"

	. "github.com/j178/leetgo/testutils/go"
)

// @lc code=begin

func twoSum(nums []int, target int) (ans []int) {
	seen := make(map[int]int)

	for i, num := range nums {
		want := target - num

		if j, ok := seen[want]; ok {
			return []int{j, i}
		}

		seen[num] = i
	}

	return
}
// @lc code=end

func main() {
	stdin := bufio.NewReader(os.Stdin)
	nums := Deserialize[[]int](ReadLine(stdin))
	target := Deserialize[int](ReadLine(stdin))
	ans := twoSum(nums, target)

	fmt.Println("\noutput:", Serialize(ans))
}

File: python/0001.two-sum/solution.py
# Created by ericklucioh at 2026/05/05 21:24
# leetgo: dev
# https://leetcode.com/problems/two-sum/

from typing import *
from leetgo_py import *

# @lc code=begin

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        n = len(nums)
        for num in nums:
            index = nums.index(num)
            j = index +1
            while j < n:
                if num + nums[j] == target:
                    return [index, j]
                j = j +1
                if j == n:
                    break
        return []
        

# @lc code=end

if __name__ == "__main__":
    nums: List[int] = deserialize("List[int]", read_line())
    target: int = deserialize("int", read_line())
    ans = Solution().twoSum(nums, target)
    print("\noutput:", serialize(ans, "integer[]"))
